extensions/agent/src/dialogs/notebookDialog.ts

```diff
@@ -0,0 +1,335 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+import * as nls from 'vscode-nls';
+import * as path from 'path';
+import * as azdata from 'azdata';
+import { JobData } from '../data/jobData';
+import { JobStepDialog } from './jobStepDialog';
+import { PickScheduleDialog } from './pickScheduleDialog';
+import { AlertDialog } from './alertDialog';
+import { AgentDialog } from './agentDialog';
+import { AgentUtils } from '../agentUtils';
+import { JobStepData } from '../data/jobStepData';
+import { NotebookData } from '../data/notebookData';
+
+
+const localize = nls.loadMessageBundle();
+
+export class NotebookDialog extends AgentDialog<NotebookData>  {
+
+	// TODO: localize
+	// Top level
+	private static readonly CreateDialogTitle: string = localize('notebookDialog.newJob', 'New Notebook Job');
+	private static readonly EditDialogTitle: string = localize('notebookDialog.editJob', 'Edit Notebook Job');
+	private readonly GeneralTabText: string = localize('notebookDialog.general', 'General');
+	private readonly BlankJobNameErrorText: string = localize('notebookDialog.blankJobNameError', 'The name of the job cannot be blank.');
+
+	// Notebook details strings
+	private readonly NotebookDetailsSeparatorTitle: string = localize('notebookDialog.notebookSection', "Notebook Details");
+	private readonly TemplateNotebookTextBoxLabel: string = localize('notebookDialog.templateNotebook', 'Notebook Path');
+	private readonly TargetDatabaseDropdownLabel: string = localize('notebookDialog.targetDatabase', 'Metadata Database');
+	private readonly ExecuteDatabaseDropdownLabel: string = localize('notebookDialog.executeDatabase', 'Execute Database');
+
+	// Job details string
+	private readonly JobDetailsSeparatorTitle: string = localize('notebookDialog.jobSection', "Job Details");
+	private readonly NameTextBoxLabel: string = localize('notebookDialog.name', 'Name');
+	private readonly OwnerTextBoxLabel: string = localize('notebookDialog.owner', 'Owner');
+	private readonly SchedulesTopLabelString: string = localize('notebookDialog.schedulesaLabel', 'Schedules list');
+	private readonly PickScheduleButtonString: string = localize('notebookDialog.pickSchedule', 'Pick Schedule');
+	private readonly ScheduleNameLabelString: string = localize('notebookDialog.scheduleNameLabel', 'Schedule Name');
+	private readonly DescriptionTextBoxLabel: string = localize('notebookDialog.description', 'Description');
+
+	// Event Name strings
+	private readonly NewJobDialogEvent: string = 'NewJobDialogOpened';
```

Comment:
Change this to specify it's a Notebook Job. Perhaps to `NewNotebookJobDialogOpened`